### PR TITLE
Document how to upgrade a Helm managed VSO

### PR DESCRIPTION
**Pull request by benashz**
Provide detailed steps for upgrading a Helm installed/managed vault-secrets-operator.

### CODE REVIEW

File: website/content/docs/platform/k8s/vso/installation.mdx [hunk 1]
```diff
@@ -24,26 +24,69 @@ HashiCorp helm repository and ensure you have access to the chart:
...
 ```shell-session
-$ helm install --create-namespace --namespace vault-secrets-operator vault-secrets-operator hashicorp/vault-secrets-operator --version 0.3.1
+$ helm install --version 0.3.1 --create-namespace --namespace vault-secrets-operator vault-secrets-operator hashicorp/vault-secrets-operator
 ```
 
 
 ## Upgrading using helm
 
-Upgrades may be performed with `helm upgrade` on an existing install. Please
+Upgrades may be performed with `helm upgrade` on an existing installation. Please
```
Comment (schavis) [inline website/content/docs/platform/k8s/vso/installation.mdx:47]:
```suggestion
You can upgrade on an existing installation with the `helm upgrade` . Please
```

Style correction: write in active voice, address the reader when providing instruction

File: website/content/docs/platform/k8s/vso/installation.mdx [hunk 2]
```diff
@@ -24,26 +24,69 @@ HashiCorp helm repository and ensure you have access to the chart:
...
 always run Helm with `--dry-run` before any install or upgrade to verify
 changes.
 
+Update the `hashicorp` Helm repo.
```
Comment (schavis) [inline website/content/docs/platform/k8s/vso/installation.mdx:51]:
```suggestion
Update the `hashicorp` Helm repo:
```
Style correction: introduce code with a colon.

File: website/content/docs/platform/k8s/vso/installation.mdx [hunk 3]
```diff
@@ -24,26 +24,69 @@ HashiCorp helm repository and ensure you have access to the chart:
...
+```shell-session
+$ helm repo update hashicorp
+
+Hang tight while we grab the latest from your chart repositories...
+...Successfully got an update from the "hashicorp" chart repository
+Update Complete. ⎈Happy Helming!⎈
+```
+
+Upgrade the VSO release
```
Comment (schavis) [inline website/content/docs/platform/k8s/vso/installation.mdx:60]:
```suggestion
Upgrade the VSO release:
```
Style correction: introduce code with a colon

File: website/content/docs/platform/k8s/vso/installation.mdx [hunk 4]
```diff
@@ -24,26 +24,69 @@ HashiCorp helm repository and ensure you have access to the chart:
...
+```shell-session
+$ helm upgrade --version 0.3.1 --namespace vault-secrets-operator vault-secrets-operator hashicorp/vault-secrets-operator
+```
+
+~> **Note:** Helm does not automatically update the Operator's CRDs.
+All CRDs will need to be updated manually after upgrading VSO.
+Refer to [Updating the CRDs](#updating-the-crds) for the proper update procedure.
```
Comment (schavis) [inline website/content/docs/platform/k8s/vso/installation.mdx:67]:
```suggestion
<Note title="Helm does not automatically update CRDs">

You must update all CRDs manually after upgrading VSO. Refer to
the  [Updating CRDs](#updating-the-crds) section for details.

</Note>
```
Style correction: write in active voice, address the reader when providing instruction, use new aside notation

File: website/content/docs/platform/k8s/vso/installation.mdx [hunk 5]
```diff
@@ -24,26 +24,69 @@ HashiCorp helm repository and ensure you have access to the chart:
...
+
+
+## Updating the CRDs
```
Comment (schavis) [inline website/content/docs/platform/k8s/vso/installation.mdx:70]:
```suggestion
## Updating CRDs
```

File: website/content/docs/platform/k8s/vso/installation.mdx [hunk 6]
```diff
@@ -24,26 +24,69 @@ HashiCorp helm repository and ensure you have access to the chart:
...
+
+Updating VSO's CRDs is a manual step that should be executed after upgrading to a newer version of the operator.
+Applicable when VSO is managed by Helm.
```
Comment (schavis) [inline website/content/docs/platform/k8s/vso/installation.mdx:73]:
```suggestion
You must update CRDs for VSO manually **after** you upgrade the
 operator when the operator is managed by Helm.
```

Style correction: write in active voice

File: website/content/docs/platform/k8s/vso/installation.mdx [hunk 7]
```diff
@@ -24,26 +24,69 @@ HashiCorp helm repository and ensure you have access to the chart:
...
+```shell-session
+$ helm show crds --version 0.3.1 hashicorp/vault-secrets-operator | kubectl apply -f -
+
+customresourcedefinition.apiextensions.k8s.io/hcpauths.secrets.hashicorp.com unchanged
+customresourcedefinition.apiextensions.k8s.io/hcpvaultsecretsapps.secrets.hashicorp.com unchanged
+customresourcedefinition.apiextensions.k8s.io/vaultauths.secrets.hashicorp.com unchanged
+customresourcedefinition.apiextensions.k8s.io/vaultconnections.secrets.hashicorp.com unchanged
+customresourcedefinition.apiextensions.k8s.io/vaultdynamicsecrets.secrets.hashicorp.com unchanged
+customresourcedefinition.apiextensions.k8s.io/vaultpkisecrets.secrets.hashicorp.com unchanged
+customresourcedefinition.apiextensions.k8s.io/vaultstaticsecrets.secrets.hashicorp.com unchanged
```
Comment (schavis) [inline website/content/docs/platform/k8s/vso/installation.mdx:87]:
````suggestion
```shell-session
$ helm show crds --version YOUR_VSO_VERSION hashicorp/vault-secrets-operator | kubectl apply -f -
```

For example, if you use VSO version 0.3.1:

```shell-session
$ helm show crds --version 0.3.1 hashicorp/vault-secrets-operator | kubectl apply -f -

customresourcedefinition.apiextensions.k8s.io/hcpauths.secrets.hashicorp.com unchanged
customresourcedefinition.apiextensions.k8s.io/hcpvaultsecretsapps.secrets.hashicorp.com unchanged
customresourcedefinition.apiextensions.k8s.io/vaultauths.secrets.hashicorp.com unchanged
customresourcedefinition.apiextensions.k8s.io/vaultconnections.secrets.hashicorp.com unchanged
customresourcedefinition.apiextensions.k8s.io/vaultdynamicsecrets.secrets.hashicorp.com unchanged
customresourcedefinition.apiextensions.k8s.io/vaultpkisecrets.secrets.hashicorp.com unchanged
customresourcedefinition.apiextensions.k8s.io/vaultstaticsecrets.secrets.hashicorp.com unchanged
````

Style correction: make placeholders obvious at a glance

File: website/content/docs/platform/k8s/vso/installation.mdx [hunk 8]
```diff
@@ -24,26 +24,69 @@ HashiCorp helm repository and ensure you have access to the chart:
...
+Refer to [Updating the CRDs](#updating-the-crds) for the proper update procedure.
+
+
+## Updating the CRDs
+
+Updating VSO's CRDs is a manual step that should be executed after upgrading to a newer version of the operator.
+Applicable when VSO is managed by Helm.
+
+~> **Note:** Make sure that the Helm chart version used in command below matches the currently
+deployed VSO release.
```
Comment (schavis) [inline website/content/docs/platform/k8s/vso/installation.mdx:76]:
```suggestion
To update your CRDs, replace `YOUR_VSO_VERSION` with your currently
 deployed VSO version:
```

Style correction: write in active voice, make placeholders obvious at a glance.

File: website/content/docs/platform/k8s/vso/installation.mdx [hunk 9]
```diff
@@ -24,26 +24,69 @@ HashiCorp helm repository and ensure you have access to the chart:
...
+customresourcedefinition.apiextensions.k8s.io/vaultdynamicsecrets.secrets.hashicorp.com unchanged
+customresourcedefinition.apiextensions.k8s.io/vaultpkisecrets.secrets.hashicorp.com unchanged
+customresourcedefinition.apiextensions.k8s.io/vaultstaticsecrets.secrets.hashicorp.com unchanged
+```
+
+## Chart values
 
-### Helm chart values
-All supported Helm chart values can be found [here](/vault/docs/platform/k8s/vso/helm)
+All supported chart values are documented [here](/vault/docs/platform/k8s/vso/helm)
```
Comment (schavis) [inline website/content/docs/platform/k8s/vso/installation.mdx:92]:
```suggestion
Refer to the [VSO helm chart](/vault/docs/platform/k8s/vso/helm)
 overview for a full list of supported chart values.
```

Style correction: use meaningful anchor text

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/docs/platform/k8s/vso/installation.mdx (hunk 7 above)
Comment (benashz) [reply website/content/docs/platform/k8s/vso/installation.mdx]: Fixed in https://github.com/hashicorp/vault/pull/23512/commits/f088661672eb9de9cea37ec46f670c893fdebd7d

File: website/content/docs/platform/k8s/vso/installation.mdx (hunk 8 above)
Comment (benashz) [reply website/content/docs/platform/k8s/vso/installation.mdx]: Fixed in https://github.com/hashicorp/vault/pull/23512/commits/f088661672eb9de9cea37ec46f670c893fdebd7d

File: website/content/docs/platform/k8s/vso/installation.mdx (hunk 4 above)
Comment (benashz) [reply website/content/docs/platform/k8s/vso/installation.mdx]: Fixed in https://github.com/hashicorp/vault/pull/23512/commits/f088661672eb9de9cea37ec46f670c893fdebd7d

Please note that Markdown syntax does not work in an HTML tag. I updated your suggestion.

File: website/content/docs/platform/k8s/vso/installation.mdx (hunk 1 above)
Comment (benashz) [reply website/content/docs/platform/k8s/vso/installation.mdx]: Fixed in https://github.com/hashicorp/vault/pull/23512/commits/f088661672eb9de9cea37ec46f670c893fdebd7d

File: website/content/docs/platform/k8s/vso/installation.mdx [hunk 10]
```diff
@@ -24,26 +24,87 @@ HashiCorp helm repository and ensure you have access to the chart:
...
+
+Hang tight while we grab the latest from your chart repositories...
+...Successfully got an update from the "hashicorp" chart repository
+Update Complete. ⎈Happy Helming!⎈
+```
+
+<note title="Helm does not automatically update CRDs">
+  You must update all CRDs manually before upgrading VSO.
+  Refer to <a href="#updating-the-crds">Updating CRDs</a>.
+</note>
```
Comment (tvoran) [inline website/content/docs/platform/k8s/vso/installation.mdx:63]: "Note" apparently needs to be capitalized to render properly, and the href needs updating.

```suggestion
<Note title="Helm does not automatically update CRDs">
  You must update all CRDs manually before upgrading VSO.
  Refer to <a href="#updating-crds">Updating CRDs</a>.
</Note>
```

File: website/content/docs/platform/k8s/vso/installation.mdx [hunk 11]
```diff
@@ -24,26 +24,87 @@ HashiCorp helm repository and ensure you have access to the chart:
...
+
 "hashicorp" has been added to your repositories
+```
 
+```shell-session
 $ helm search repo hashicorp/vault-secrets-operator
+
 NAME           	CHART VERSION	APP VERSION	DESCRIPTION
 hashicorp/vault-secrets-operator	0.3.1       	0.3.1     	Official HashiCorp Vault Secrets Operator Chart
 ```
```
Comment (tvoran) [inline website/content/docs/platform/k8s/vso/installation.mdx:36]: Just my personal preference but the extra lines look a little odd to me here:

````suggestion
```shell-session
$ helm repo add hashicorp https://helm.releases.hashicorp.com
"hashicorp" has been added to your repositories
```

```shell-session
$ helm search repo hashicorp/vault-secrets-operator
NAME           	CHART VERSION	APP VERSION	DESCRIPTION
hashicorp/vault-secrets-operator	0.3.1       	0.3.1     	Official HashiCorp Vault Secrets Operator Chart
```
````

**Review by tvoran — APPROVED** [review]
The procedure looks good, and worked for me going from v0.2.0 -> v0.3.1. Just a fix for the `note` syntax.

**Review by kschoche — APPROVED** [review]

File: website/content/docs/platform/k8s/vso/installation.mdx (hunk 11 above)
Comment (benashz) [reply website/content/docs/platform/k8s/vso/installation.mdx:36]: Fixed with 9740993ca9c417385bd16128de261574c6515df3

**Review by schavis — CHANGES_REQUESTED** [review]

**Comment by schavis** [comment]
> @schavis would you mind taking another look?

One last, minor change, otherwise it looks good. Feel free to ping me in Slack to approve once it's updated

File: website/content/docs/platform/k8s/vso/installation.mdx [hunk 12]
```diff
@@ -34,16 +36,72 @@ hashicorp/vault-secrets-operator	0.3.1       	0.3.1     	Official HashiCorp Vaul
...
+$ helm show crds --version 0.3.1 hashicorp/vault-secrets-operator | kubectl apply -f -
+$ helm upgrade --version 0.3.1 --namespace vault-secrets-operator vault-secrets-operator hashicorp/vault-secrets-operator
+```
+
+## Updating CRDs
+
+You must update the CRDs for VSO manually **before** you upgrade the
+ operator when the operator is managed by Helm.
+
+*Any kubectl warnings related to `last-applied-configuration` should be safe to ignore.*
```
[the remark this answers is not in the document]
Comment (benashz) [reply website/content/docs/platform/k8s/vso/installation.mdx]: Fixed with 477a080ec56ba7a54aac5f1f5cd688a6c36d2ca1

**Review by schavis — APPROVED** [review]

**Pull request merged**